From 9247f9567e432755f4a1b6236c9638a7a2723250 Mon Sep 17 00:00:00 2001
From: rramasamy <rramasamy@maxlinear.com>
Date: Tue, 3 Jun 2025 11:30:45 +0800
Subject: [PATCH] agent: son_slave_thread: Agent sends rssi instead of rcpi

The Agent was incorrectly sending RSSI instead of RCPI in the Unassociated
STA Link Metrics Response TLV. This update aligns with the EasyMesh
specification by converting RSSI to RCPI using the existing
convert_rcpi_from_rssi API.

Closes PPM-3331

Signed-off-by: rramasamy <rramasamy@maxlinear.com>
---
 agent/src/beerocks/slave/son_slave_thread.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/agent/src/beerocks/slave/son_slave_thread.cpp b/agent/src/beerocks/slave/son_slave_thread.cpp
index 6e1494037f..e873ede21a 100644
--- a/agent/src/beerocks/slave/son_slave_thread.cpp
+++ b/agent/src/beerocks/slave/son_slave_thread.cpp
@@ -3675,9 +3675,10 @@ bool slave_thread::handle_cmdu_monitor_message(const std::string &fronthaul_ifac
             }
 
             stations_stats one_station_stats;
-            one_station_stats.channel_number      = station_in.channel;
-            one_station_stats.uplink_rcpi_dbm_enc = station_in.signal_strength;
-            one_station_stats.sta_mac             = station_in.sta_mac;
+            one_station_stats.channel_number = station_in.channel;
+            one_station_stats.uplink_rcpi_dbm_enc =
+                wireless_utils::convert_rcpi_from_rssi(station_in.signal_strength);
+            one_station_stats.sta_mac = station_in.sta_mac;
             one_station_stats.measurement_to_report_delta_msec =
                 (uint32_t)std::chrono::duration_cast<std::chrono::milliseconds>(
                     std::chrono::steady_clock::now().time_since_epoch() -
-- 
GitLab

